### PR TITLE
Clytwynec/removing rake install on jenkins

### PR DESCRIPTION
**Pull request by clytwynec**
@jzoldak @benpatterson  Please review...

Is there anyone else that should review this? 

I've tested that devstack is fine with this (but there weren't really any changes to the devstack config).  How would I go about checking this in Jenkins?

This goes with https://github.com/edx/edx-platform/pull/4634

### CODE REVIEW

**Comment by jzoldak** [comment]
someone from devops. @hkim823 ?

**Comment by hkim823** [comment]
I'm fine with this, I wonder if https://github.com/edx/configuration/blob/master/playbooks/roles/rbenv/defaults/main.yml should also remove the rbenv_rake_version variable.

**Comment by clytwynec** [comment]
@hkim823  I think that var will  still be needed for the forums rbenv

**Comment by hkim823** [comment]
Makes sense to me :+1:

**Comment by feanil** [comment]
I think we should also test this on a single instance sandbox is possible.

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 1]
```diff
@@ -89,11 +89,11 @@
...
   - "restart edxapp_workers"
 
-# Gather assets using rake if possible
+# Gather assets using paver if possible
 
-- name: gather {{ item }} static assets with rake
+- name: gather {{ item }} static assets with paver
   shell: >
-    SERVICE_VARIANT={{ item }} rake {{ item }}:gather_assets:aws
+    SERVICE_VARIANT={{ item }} paver update_assets {{ system }} --settings=aws
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: Where does the 'system' var get set?

Comment (clytwynec) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: ahhh. That was a typo, it should be {{ item }} as well.  Will fix.

Comment (hkim823) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: Good point, on a single instance sandbox {{ system }} is undefined.

**Comment by clytwynec** [comment]
How do I go about testing this on a sandbox instance?

**Comment by hkim823** [comment]
This is happy in my single instance sandbox @feanil

**Comment by hkim823** [comment]
@clytwynec i just did a `sudo /edx/bin/update configuration clytwynec/removing_rake_install_on_jenkins && sudo /edx/bin/update edx-platform master` and things were happy with the recent change

Lastly on http://jenkins.edx.org:8080/job/ansible-provision you could probably change the parameters to build with your branch of configuration and see if it completes properly

**Comment by clytwynec** [comment]
Thanks @hkim823 !

**Comment by feanil** [comment]
Looks good to me.

**Comment by clytwynec** [comment]
@jzoldak  @benpatterson  I've tested this via the manual testing process described [here](https://edx-wiki.atlassian.net/wiki/display/ENG/Jenkins+TestEng+Administration).  The tests are passing on the new instance and it has the expected changes in provisioning.   Is it ok to merge this and update the AMI-ID on jenkins? 

After that, I can merge https://github.com/edx/edx-platform/pull/4634.

**Comment by benpatterson** [comment]
I say you're good to go. If there is no time pressure on it, I would wait until tomorrow morning to make the Jenkins Master change since load is generally a bit lower and you can more easily see how the new worker does. 

Regardless of timing, when you do make the change, you can manually provision a jenkins-worker by going to manage nodes and clicking the provision dropdown. Might want to do that so you easily know which node is running off of the new AMI (AWS console will tell you...but just to shortcut the lookup). Plus  you don't have to wait until Jenkins decides to provision a new one.

**Comment by benpatterson** [comment]
:+1: by the way

**Comment by jzoldak** [comment]
:thumbsup:

Along the lines of @benpatterson 's comment - After changing the AMI ID for workers in the jenkins config, the easiest way to get rid of the old nodes gracefully is to go to each of them and use "Mark this node temporarily offline." That way no new builds will run on the old node, and after it is idle for 5 minutes it'll automatically terminate on aws.

**Pull request merged**